engine.py

```diff
@@ -515,52 +510,112 @@ def has_ui(self):
     ##########################################################################################
     # logging
 
-    def log_debug(self, msg):
+    def _emit_log_message_FUTURE(self, handler, record):
         """
-        Log debug to the Maya script editor
+        Called by the engine to log messages in Maya script editor.
+        All log messages from the toolkit logging namespace will be passed to this method.
+
+        .. note:: This method will be called by the new logging system introduced in core v0.18.
+                  It will be enabled in January 2017 by removing the method name suffix "_FUTURE".
 
-        :param msg: The message to log
+        :param handler: Log handler that this message was dispatched from.
+        :type handler: :class:`~python.logging.LogHandler`
+        :param record: Standard python logging record.
+        :type record: :class:`~python.logging.LogRecord`
         """
-        global g_last_message_time
 
-        # get the time stamps
-        prev_time = g_last_message_time
-        current_time = time.time()
+        # Do not log debug messages when debug logging setting is off.
+        if record.levelno < logging.INFO and not self.get_setting("debug_logging", False):
+            return
+
+        # Give a standard format to the message.
+        msg = "Shotgun %s" % handler.format(record)
+
+        # Select Maya display function to use according to the logging record level.
+        if record.levelno < logging.WARNING:
+            fct = OpenMaya.MGlobal.displayInfo
+        elif record.levelno < logging.ERROR:
+            fct = OpenMaya.MGlobal.displayWarning
+        else:
+            fct = OpenMaya.MGlobal.displayError
 
-        # update global counter
-        g_last_message_time = current_time
+        # Display the message in Maya script editor in a thread safe manner.
+        self.async_execute_in_main_thread(fct, msg)
 
+    def log_debug(self, msg):
+        """
+        Logs debug messages in Maya script editor.
+
+        .. note:: This method is called by the legacy logging system.
```

Comment:
I think it is a good thing to mentioned in advance changes that we know will be done in the code. So, instead of removing the note, I updated it to answer the kind of concerns you expressed.